cli-toolchain/graphql.md

```diff
@@ -1694,14 +1694,221 @@ The `@function` directive generates these resources as necessary:
+
+Likewise, you can make a simple one-to-many connection as follows for a post that has many comments:
+
+**Has Many**
+
+```
+type Post @model {
+    id: ID!
+    title: String!
+    comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+}
+
+type Comment
+    @model
+    @key(name: "byPost", fields: ["postID", "content"])
+{
+    id: ID!
+    postID: ID!
+    content: String!
+}
+```
+
+Note how a one-to-many connection needs a @key that allows comments to be queried by the postID and the connection uses this key to get all comments whose postID is the id of the post was called on.
+After it's transformed, you can create comments and query the connected Post as follows:
+
+```
+mutation CreateCommentOnPost {
+    createComment(input: { content: "A comment", postID: "a-post-id"}) {
+        id
+        content
+    }
+}
+```
+
+And you can query a Post with its comments as follows:
+
+```
+query getPost {
+    getPost(input: { id: "a-post-id" }) {
+        id
+        title
+        comments {
+          items {
+              id
+              content
+          }
+        }
+    }
+}
+```
+**Belongs To**
+
+You can make a connection bi-directional by adding a many-to-one connection to types that already have a one-to-many connection. In this case we add a connection from Comment to Post since each comment belongs to a post:
+
+```
+type Post @model {
+    id: ID!
+    title: String!
+    comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+}
+
+type Comment
+    @model
+    @key(name: "byPost", fields: ["postID", "content"])
+{
+    id: ID!
+    postID: ID!
+    content: String!
+
+    post: Post @connection(fields: ["postID"])
+}
+```
+After it's transformed, you can create comments with a post as follows:
+
+```
+mutation CreateCommentOnPost {
+    createComment(input: { content: "A comment", postID: "a-post-id"}) {
+        id
+        content
+        post {
+            id
+            title
+            comments
+        }
+    }
+}
+```
+
+**Many-To-Many Connections**
+
+You can implement many to many yourself using two 1-M @connections and a joining @model. For example:
+
+```
+type Post @model {
+    id: ID!
+    title: String!
+    editors: [PostEditor] @connection(keyName: "byPost", fields: ["id"])
+}
+
+# Create a join model and disable queries as you don't need them
+# and can query through Post.editors and User.posts
+type PostEditor
+    @model(queries: null)
+    @key(name: "byPost", fields: ["postID", "editorID"])
+    @key(name: "byEditor", fields: ["editorID", "postID"])
+{
+    id: ID!
+    postID: ID!
+    editorID: ID!
+    post: Post! @connection(fields: ["postID"])
+    editor: User! @connection(fields: ["editorID"])
+}
+
+type User @model {
+    id: ID!
+    username: String!
+    posts: [PostEditor] @connection(keyName: "byEditor", fields: ["byEditor"])
```

Comment:
Is the fields argument in the @connection directive supposed to be: `fields: ["id"]` instead of `fields: ["byEditor"]`?